arch/x86/core/intel64/cpu.c

```diff
@@ -108,7 +144,20 @@ FUNC_NORETURN void z_x86_cpu_init(struct x86_cpuboot *cpuboot)
 	z_x86_msr_write(X86_FMASK_MSR, EFLAGS_SYSCALL);
 #endif
 
+	/* Cache AP info. */
+	__ASSERT(!x86_update_cpu_info(cpu_num), "Get AP CPU info failed");
```

Comment:
Call this inside ASSERT is not the best way, are you sure ASSERT is enabled? Better check return code.